pkg/services/datasources/service/store.go

```diff
@@ -377,6 +379,10 @@ func (ss *SqlStore) UpdateDataSource(ctx context.Context, cmd *datasources.Updat
 			}
 		}
 
+		if !util.IsValidShortUID(cmd.UID) {
+			logDeprecatedInvalidDsUid(ss.logger, cmd.UID, cmd.Name)
+		}
```

Comment:
I wonder, if the UID is invalid, would the code reach this or would fail before? Maybe it's better to move this warning to the top of the function?

Reply:
This block is executed after committing to the database, so it will be executed only if the changes are committed. There's no validation before updating the UID in the database, so the update will not fail either